src/frontend/planner_test/tests/testdata/input/temporal_filter.yaml

```diff
@@ -23,6 +23,13 @@
   expected_outputs:
   - stream_plan
   - stream_dist_plan
+- name: Temporal filter with equal condition
+  sql: |-
+    create table t1 (ts timestamp with time zone);
+    select * from t1 where date_trunc('week', now())  = date_trunc('week',ts);
```

Comment:
When the condition changes, it looks like we will generate a large amount of data within a single barrier.

Reply:
Yes, but we do not ban it for other hash join or dynamic filter with the scalar subquery, so at least we need to support it and solve those common issue in another way.

Reply:
Yes, but we do not ban it for other hash join or dynamic filter with the scalar subquery, so at least we need to support it and solve that common issue in another way.

Reply:
Yes, but we do not ban it for other hash join or dynamic filter with the scalar subquery, so at least we need to support it and solve that common issue in another way.

Reply:
I am not sure. This condition somehow looks like a nested loop join or no partition over window cc @fuyufjh

Reply:
In other words if we allow this query, we should allow this too...
```
dev=> explain create materialized view mv as 
select * from t1 where date_trunc('week', ts)  = (select max(date_trunc('week', ts)) from t1);
NOTICE:  Your session timezone is UTC. It was used in the interpretation of timestamps and dates in your query. If this is unintended, change your timezone to match that of your data's with `set timezone = [timezone]` or rewrite your query with an explicit timezone conversion, e.g. with `AT TIME ZONE`.

                                                                            QUERY PLAN                                                                             
-------------------------------------------------------------------------------------------------------------------------------------------------------------------
 StreamMaterialize { columns: [ts, t1._row_id(hidden), $expr1(hidden)], stream_key: [t1._row_id, $expr1], pk_columns: [t1._row_id, $expr1], pk_conflict: NoCheck }
 └─StreamExchange { dist: HashShard(t1._row_id, $expr1) }
   └─StreamHashJoin { type: Inner, predicate: $expr1 = max(max($expr2)) }
     ├─StreamExchange { dist: HashShard($expr1) }
     │ └─StreamProject { exprs: [t1.ts, DateTrunc('week':Varchar, t1.ts, 'UTC':Varchar) as $expr1, t1._row_id] }
     │   └─StreamTableScan { table: t1, columns: [ts, _row_id] }
     └─StreamExchange { dist: HashShard(max(max($expr2))) }
       └─StreamProject { exprs: [max(max($expr2))] }
         └─StreamSimpleAgg { aggs: [max(max($expr2)), count] }
           └─StreamExchange { dist: Single }
             └─StreamHashAgg { group_key: [$expr3], aggs: [max($expr2), count] }
               └─StreamProject { exprs: [DateTrunc('week':Varchar, t1.ts, 'UTC':Varchar) as $expr2, t1._row_id, Vnode(t1._row_id) as $expr3] }
                 └─StreamTableScan { table: t1, columns: [ts, _row_id] }
(13 rows)```

Reply:
Agree that we should support this... Even though it's very likely to become a performance issue... We may warn the users explicitly.

But I am confused by the title: "temporal filter with complex expression". Here you actually run the query with `HashJoin (left-semi)`, and it has nothing to do with temporal filter, right?

Reply:
Yes, but in the user's view, what he write in the SQL is a "temporal filter". SemiJoin or dynamic filter are just internal implementations.

Reply:
I see. Additionally, for "temporal filter", I would also expect range conditions e.g.

```
select * from t1 where date_trunc('minute', now()) - interval '5 minutes' <= date_trunc('minute', ts);
```

Shall we unify them with unique implementation?

Reply:
https://github.com/risingwavelabs/risingwave/issues/14085

Reply:
> I see. Additionally, for "temporal filter", I would also expect range conditions e.g.
> 
> ```
> select * from t1 where date_trunc('minute', now()) - interval '5 minutes' <= date_trunc('minute', ts);
> ```
> 
> Shall we unify them with unique implementation?

#14085